server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -637,27 +631,79 @@ private List<String> createClusterChain(final Map<String, ClusterMetadataManifes
             return Collections.emptyList();
         }
         if (validClusterUUIDs.size() > 1) {
-            throw new IllegalStateException(
-                String.format(
-                    Locale.ROOT,
-                    "The system has ended into multiple valid cluster states in the remote store. "
-                        + "Please check their latest manifest to decide which one you want to keep. Valid Cluster UUIDs: - %s",
-                    validClusterUUIDs
-                )
+            final Map<String, ClusterMetadataManifest> manifestsByClusterUUIDTrimmed = trimClusterUUIDs(
+                manifestsByClusterUUID,
+                validClusterUUIDs,
+                clusterName
             );
+            if (manifestsByClusterUUID.size() == manifestsByClusterUUIDTrimmed.size()) {
+                throw new IllegalStateException(
+                    String.format(
+                        Locale.ROOT,
+                        "The system has ended into multiple valid cluster states in the remote store. "
+                            + "Please check their latest manifest to decide which one you want to keep. Valid Cluster UUIDs: - %s",
+                        validClusterUUIDs
+                    )
+                );
+            }
+            return createClusterChain(manifestsByClusterUUIDTrimmed, clusterName);
         }
         final List<String> validChain = new ArrayList<>();
         String currentUUID = validClusterUUIDs.get(0);
-        while (!ClusterState.UNKNOWN_UUID.equals(currentUUID)) {
+        while (currentUUID != null && !ClusterState.UNKNOWN_UUID.equals(currentUUID)) {
             validChain.add(currentUUID);
             // Getting the previous cluster UUID of a cluster UUID from the clusterUUID Graph
             currentUUID = clusterUUIDGraph.get(currentUUID);
         }
         return validChain;
     }
 
+    private Map<String, ClusterMetadataManifest> trimClusterUUIDs(
+        final Map<String, ClusterMetadataManifest> manifestsByClusterUUID,
+        final List<String> validClusterUUIDs,
+        final String clusterName
+    ) {
+        final Map<String, ClusterMetadataManifest> manifestsByClusterUUIDTrimmed = new HashMap<>(manifestsByClusterUUID);
```

Comment:
name -> trimmedUUIDs 
// comment - In case currentUUID's latest manifest is same as previous UUIDs latest manifest. That means it was restored from previousUUID and no IndexMetadata update was performed on it.